src/services/ban_service.py

```diff
@@ -3,66 +3,62 @@
+     Args:
+         invoking_member (discord.Member): The moderator initiating the ban.
+         user (discord.Member): The user being banned.
+         reason (str): Reason for the ban.
+         delete_messages_flag (bool): Whether to delete the user's messages.
 
     Returns:
-        Optional[Tuple[bool, bool, str]]: tuple containing (ban_result,dm_state,result_description)
-            - ban_result (bool): Indicates whether the ban succeeded.
-            - dm_state (bool): Indicates whether the DM notification to the user succeeded.
-            - result_description (str): A message describing the status of the ban operation.
+         bool: status of the ban
+         str: any errors that occurred
     """
-    if len(reason) < 512:
-        result_description = ""
-        dm_state = False
-        # attempt to send a DM to the user with the reason for the ban
-        try:
-            # TO DO: When the appeal process is implemented, add a link to the appeal process in the message.
-            await send_dm(
-                user,
-                f"You are being banned from NA Ultimate Raiding - FF XIV for the following reason: \n> {reason} "
-                "\nYou may appeal this ban by contacting the moderators of the server in 30 days.",
-            )
-            logger.info(f"Successfully sent dm to {user.mention}")
-            dm_state = True
-        except Exception as e:
-            logger.error(f"DM to {user.mention} due failed due to error: {e}")
-            dm_state = False
+    if len(reason) >= 512:
+        return (
+            False,
+            f"Unable to ban {user.mention}: reason is too long (above 512 characters). Please shorten the ban reason.",
+        )
 
-        # attempt Ban
-        try:
-            await user.ban(reason=reason)
-            logger.info(f"Successfully banned {user.mention}")
-            ban_result = True
-        except Exception as e:
-            logger.error(f"Ban of {user.mention} failed due to error: {e}")
-            ban_result = False
+    # Calculate the timestamp for 30 days from now
+    appeal_deadline = int((datetime.now(timezone.utc) + timedelta(days=30)).timestamp())
 
-        if not ban_result and dm_state:  # ban fail dm succeed.
-            result_description = f"Unable to ban {user.mention}, please ban via discord built-in ban feature. A DM has been sent to {user.mention} with ban reason."
-        elif ban_result and not dm_state:  # ban succeed dm fail.
-            result_description = (
-                f"Successfully banned {user.mention} but DM failed to send."
-            )
-        elif ban_result and dm_state:  # full success
-            result_description = f"Successfully banned {user.mention} and DM has been sent with ban reason."
-        else:  # full failure
-            result_description = f"Unable to ban {user.mention} and unable to send DM. Please ban via discord built-in ban feature or try again later."
+    dm_failed = None
```

Comment:
let's rename this to `dm_failed_message` just for clarity

Reply:
done